### PR TITLE
(#3427) Rework build to remove use of -unpackself

### LINKED ISSUE #3427: Rework build process to remove the use of `-unpackself`
### Checklist

- [X] I have verified this is the correct repository for opening this issue.
- [X] I have verified no other issues exist related to my request.

### Is Your Feature Request Related To A Problem? Please describe.

Per #3426, we are looking to deprecate `-unpackself`. We will need to replace it in our build processes.

### Describe The Solution. Why is it needed?

Replace usage of `-unpackself` to generate the directory structure from the `chocolatey.resources` assembly with some straightforward file copies from the project into the nuget/chocolatey/choco directory.

### Additional Context

_No response_

### Related Issues

- #3426

### PR DESCRIPTION
**Pull request by vexx32**
## Description Of Changes

- Copy resource files to output directories during build
- Alter build to copy these files directly instead of using `-unpackself`
- Cleaned up a couple csproj files that had duplicated property-setting blocks for some build configurations that I noticed while fiddling with them.

## Motivation and Context

This command is deprecated and slated to be removed in v3 so we need to not be using it.

## Testing

1. Run `.\build.ps1 -Configuration ReleaseOfficial`
2. Inspect the generated nupkgs, compare to a previous release to ensure all the files seem to be in their proper places.
3. Copy the generated nupkgs to a fresh VM and then take a snapshot.
4. In a PowerShell admin shell, run the folllowing to install Chocolatey, using the path to the chocolatey nupkg:
    ```
    ${function:Install-Chocolatey} = [scriptblock]::Create((iwr https://chocolatey.org/install.ps1))
    Install-Chocolatey -ChocolateyDownloadUrl ./path/to/chocolatey.nupkg
    ```
5. Try installing, upgrading, and uninstalling a basic package and make sure all works as expected. (I used the `vlc.install` package for these).
6. Revert the VM to the snapshot from step 3
7. Repeat tests from step 4 with the no-7zip package instead. Note that any packages using `Get-ChocolateyUnzip` will likely not work.

### Operating Systems Testing

Win10

## Change Types Made

* [ ] Bug fix (non-breaking change).
* [x] Feature / Enhancement (non-breaking change).
* [ ] Breaking change (fix or feature that could cause existing functionality to change).
* [ ] Documentation changes.
* [ ] PowerShell code changes.

## Change Checklist

* [ ] Requires a change to the documentation.
* [ ] Documentation has been updated.
* [ ] Tests to cover my changes, have been added.
* [x] All new and existing tests passed?
* [ ] PowerShell code changes: PowerShell v3 compatibility checked?

## Related Issue

Fixes #3427

### CODE REVIEW

File: src/chocolatey.resources/chocolatey.resources.csproj [hunk 1]
```diff
@@ -64,98 +64,206 @@
...
-    <EmbeddedResource Include="helpers\functions\Install-ChocolateyVsixPackage.ps1" />
-    <EmbeddedResource Include="helpers\functions\Install-ChocolateyZipPackage.ps1" />
-    <EmbeddedResource Include="helpers\functions\Start-ChocolateyProcessAsAdmin.ps1" />
-    <EmbeddedResource Include="helpers\functions\Uninstall-ChocolateyPackage.ps1" />
-    <EmbeddedResource Include="helpers\functions\UnInstall-ChocolateyZipPackage.ps1" />
-    <EmbeddedResource Include="redirects\choco.exe.ignore" />
-    <EmbeddedResource Include="tools\checksum.exe.ignore" />
-    <EmbeddedResource Include="tools\shimgen.exe.ignore" />
+    <EmbeddedResource Include="helpers\chocolateyInstaller.psm1">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
Comment (gep13) [inline src/chocolatey.resources/chocolatey.resources.csproj:68]: @vexx32 why was this type of change necessary? Did you notice something during putting this PR together that meant that this was necessary?

If so, I think we need to capture this information into the commit message, for our future selves.

Comment (vexx32) [reply src/chocolatey.resources/chocolatey.resources.csproj:68]: Added some words to the commit message to make this clear.

**Review by corbob — APPROVED** [review]
This looks good to me and passes the tests that I threw at it.

**Comment by corbob** [comment]
@vexx32 as discussed in Slack: I'll leave this PR for @gep13 to respond to the thread above and will merge it after he's happy.

**Comment by corbob** [comment]
Thanks for getting this updated @vexx32

**Pull request merged**